lib/mini_profiler/profiler.rb

```diff
@@ -538,7 +538,8 @@ def dump_exceptions(exceptions)
 
         body << "\nBacktraces\n"
         exceptions.each_with_index do |e, i|
-          body << "##{i + 1}: #{e.class} - \"#{e.message}\"\n  #{e.backtrace.join("\n  ")}\n\n"
+          message = Gem::Version.new(RUBY_VERSION) < Gem::Version.new("3.1") ? e.message : e.message.lines.first.chomp
+          body << "##{i + 1}: #{e.class} - \"#{message}\"\n  #{e.backtrace.join("\n  ")}\n\n"
```

Comment:
Thank you for this! I noticed the tests were failing on ruby 3.1 and I meant to debug it but got distracted with other things!

I wonder if we should just take the first line of the error message unconditionally? Because the way exceptions are formatted is that each exception gets a "header" line that contains the exception number, class and message and then the header line is followed by the backtrace. If we get an exception where the message is multi-line, then the header for that exception becomes 2 lines and it looks weird.